File: easy/2784-check-if-array-is-good/solution.py
from collections import Counter
from typing import List


class Solution:
    def isGood(self, nums: List[int]) -> bool:
        """
        2784. Check if Array is Good
        Time: O(n)
        Space: O(n)
        """
        n = max(nums)

        if len(nums) != n + 1:
            return False

        count = Counter(nums)

        for value in range(1, n):
            if count[value] != 1:
                return False

        return count[n] == 2

File: easy/2784-check-if-array-is-good/solution.java
import java.util.HashMap;
import java.util.Map;

/**
 * 2784. Check if Array is Good
 * Time: O(n)
 * Space: O(n)
 */
class Solution {
    public boolean isGood(int[] nums) {
        int n = 0;
        for (int num : nums) {
            n = Math.max(n, num);
        }

        if (nums.length != n + 1) {
            return false;
        }

        Map<Integer, Integer> count = new HashMap<>();
        for (int num : nums) {
            count.put(num, count.getOrDefault(num, 0) + 1);
        }

        for (int value = 1; value < n; value++) {
            if (count.getOrDefault(value, 0) != 1) {
                return false;
            }
        }

        return count.getOrDefault(n, 0) == 2;
    }
}
